include/util.h

```diff
@@ -111,6 +111,8 @@
 
 #define UNUSED_PARAM(x) ((void)x)
 
+#define shift_right(x, s) ((x) < 0 ? -(-(x) >> (s)) : (x) >> (s))
```

Comment:
If the variable is signed, isn't this done by default?

Reply:
no, nothing guarantees you that

Reply:
It seems to be compiler dependent (https://en.wikipedia.org/wiki/Bitwise_operations_in_C#Right_shift_%3E%3E) 
Good to know :)